_integration-test/error-handling.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+set -ex
+
+source "$(dirname $0)/../install/_lib.sh"
+
+export REPORT_SELF_HOSTED_ISSUES=1
+
+source ../install/error-handling.sh
+
+send_envelope() {
+  echo "Test Sending $1"
+}
+
+export -f send_envelope
+echo "${_group}Testing initial send_event"
+export log_file="test_log.txt"
+echo "Test Logs" > "$basedir/$log_file"
+SEND_EVENT_RESPONSE=$(send_event "12345123451234512345123451234512" Test exited with status 1)
+if [ "$SEND_EVENT_RESPONSE" != 'Test Sending sentry-envelope-12345123451234512345123451234512' ]; then
+  exit -1
+fi
+ENVELOPE_CONTENTS=$(cat /tmp/sentry-envelope-12345123451234512345123451234512)
+if [ "$ENVELOPE_CONTENTS" !='{"event_id":"12345123451234512345123451234512","dsn":"https://19555c489ded4769978daae92f2346ca@self-hosted.getsentry.net/3"}
+{"type":"event"}
+{"message":"Test","level":"error"}
+{"type":"attachment","length":10,"content_type":"text/plain","filename":"install_log.txt"}
+Test Logs' ]; then
+  exit -1
+fi
+echo "Pass."
+echo "${_endgroup}"
+
+echo "${_group}Testing send_event duplicate"
+SEND_EVENT_RESPONSE=$(send_event "12345123451234512345123451234512" Test exited with status 1)
+if [ "$SEND_EVENT_RESPONSE" != 'Looks like you'\''ve already sent this error to us, we'\''re on it :)' ]; then
+  exit -1
+fi
+echo "Pass."
+rm "$basedir/$log_file"
+echo "${_endgroup}"
+
+echo "${_group} Testing cleanup without minimizing downtime"
```

Comment:
I'm not a huge fan of these test cases because technically I'm testing cleanup when there isn't an error code being thrown. It seems to be impossible to unit test by setting bash internal variables by overwriting `BASH_COMMAND`, `BASH_SOURCE`, `BASH_LINENO`, etc so don't think there's a better way to do this